source/gfoidl.Stochastics/Partitioners/TrapezeWorkloadPartitioner.cs

```diff
@@ -0,0 +1,127 @@
+     *  Thus the calculations are (greatly) simplified.
+     */
+    internal class TrapezeWorkloadPartitioner : WorkloadPartitioner
+    {
+        private readonly double _lambda;
+        //---------------------------------------------------------------------
+        public TrapezeWorkloadPartitioner(
+            int    size,
+            double lambda,
+            int?   partitionCount = null)
+            : base(size, partitionCount)
+            => _lambda = lambda;
+        //---------------------------------------------------------------------
+        public TrapezeWorkloadPartitioner(
+            int    size,
+            double loadFactorAtStart,
+            double loadFactorAtEnd,
+            int?   partitionCount = null)
+            : this(size, (loadFactorAtEnd - loadFactorAtStart) / size, partitionCount)
+        { }
+        //---------------------------------------------------------------------
+        protected override IEnumerable<KeyValuePair<long, Range>> GetOrderableDynamicPartitions(int partitionCount)
+            => new TrapezeWorkloadPartitions(_size, _lambda, partitionCount);
+        //---------------------------------------------------------------------
+        private class TrapezeWorkloadPartitions : Partitions
+        {
+            private readonly Range[] _partitions;
+            //-----------------------------------------------------------------
+            public TrapezeWorkloadPartitions(
+                int    size,
+                double lambda,
+                int    partitionCount)
+                : base(size, partitionCount)
+            {
+                _partitions = lambda == 0
+                    ? Rectangle()
+                    : Triangle();
+                //-------------------------------------------------------------
+                Range[] Triangle()
+                {
+                    bool increasing = true;
+                    if (lambda < 0)
+                    {
+                        increasing = false;
+                        lambda     = -lambda;
+                    }
+
+                    var partitions       = new Range[partitionCount];
+                    double triangleArea  = (double)size * size * lambda * 0.5;
+                    double partitionArea = triangleArea / partitionCount;
+                    double lambdaInv     = 1d / lambda;
+                    int start            = 0;
+
+                    for (int i = 0; i < partitions.Length; ++i)
+                    {
+                        int end;
+
+                        if (i == partitions.Length - 1)
+                            end = _size;
+                        else
+                        {
+                            double area = partitionArea * (i + 1);
+                            double tmp  = Sqrt(2 * area * lambdaInv);
+                            end         = (int)Floor(tmp);
```

Comment:
Is `Floor` the best option? The actual "int-Area" should be as close as possible to the exact area, so that the work distribution is as equal as possible.